Cargo.toml

```diff
@@ -63,6 +63,20 @@ exclude = [
   "src/tools/x",
 ]
 
+# These lints are applied to many crates in the workspace. In practice, this is
+# all crates under `compiler/`.
+#
+# NOTE: rustc-specific lints (e.g. `rustc::internal`) aren't supported by
+# Cargo. (Support for them is possibly blocked by #44690 (attributes for
+# tools).) Those lints are instead specified for `compiler/` crates in
+# `src/bootstrap/src/core/builder/cargo.rs`.
+[workspace.lints.rust]
+# FIXME(edition_2024): Change this to `-Wrust_2024_idioms` when all of the
+# individual lints are satisfied.
+keyword_idents_2024 = "warn"
+unreachable_pub = "warn"
+unsafe_op_in_unsafe_fn = "warn"
```

Comment:
Question: one more thing -- should we document (even just a brief mention about this) how to adjust non-tool lints for all compiler crates somewhere in rustc-dev-guide? (And mention that tool lints will have to be adjusted in bootstrap instead.)

Reply:
Maybe? I don't know what section it would go in.

Reply:
Since we'll probably want to do something more consistently eventually, this comment is probably good enough.